surface/include/pcl/surface/3rdparty/opennurbs/opennurbs_array_defs.h

```diff
@@ -1372,7 +1372,7 @@ void ON_ClassArray<T>::Insert( int i, const T& x )
     DestroyElement( m_a[m_count] );
 	  m_count++;
     if ( i < m_count-1 ) {
-      Move( i+1, i, m_count-1-i );
+      Move( i+1, i, (unsigned int) m_count-1-i );
```

Comment:
Some parenthesis and static_cast maybe

Reply:
What warning/error does it gives?
m_count is int, 1 is int? and i is int - given to a method "move", which accepts an int?
And casting a int to unsigned int, which is then parsed to a method that accepts a int, doesn't seem to make sense?

The same Move method is called on line 492 as well as 1403 - but those doesn't require the cast?

Reply:
> What warning/error does it gives?

 error: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Werror=strict-overflow]

> m_count is int, 1 is int? and i is int - given to a method "move", which accepts an int?
> And casting a int to unsigned int, which is then parsed to a method that accepts a int, doesn't seem to make sense?

By casting m_count to unsigned int, i and 1 also get cast to unsigned get added then cast to int from which the compiler can infer that the ele_cnt argument in Move function will always be positive.

> The same Move method is called on line 492 as well as 1403 - but those doesn't require the cast?
The warnings come due to the memory functions being called in GetCapacity and Move.

Yeah in line 492 and 1403, the dest_i argument of Move function, is set to i whereas in Insert it is i + 1. So if i = 0, then when in Insert, dest_i = i+ 1 and the condition `if ( capacity > m_capacity )` becomes true as m_capacity is 0 by default so the setCapacity function gets called which contains `memset`.  
Not sure exactly what all optimizations the compiler is doing to create such a situation

This issue can be looked into more extensively but that would better be better suited for a separate issue. The current solution solves the warning and hopefully till my understanding doesn't cause any other issues.

Reply:
> Some parenthesis and static_cast maybe

Done

Reply:
Since we are in the conditional branch ` if ( i < m_count-1 )` we are at least sure that  `m_count-1-i` is positive. But this is indeed weird af.